backend-config/account_association.go

```diff
@@ -21,6 +24,50 @@ func (c *ConfigT) processAccountAssociations() {
 	}
 }
 
+// getAccountDefinition returns a pointer to the account definition for the given account definition name.
+// If the account definition name is empty or the account definition doesn't exist, it returns nil.
+//
+// Parameters:
+//   - accountDefinitionName: The name of the account definition to look up
+//   - accountAssociationLogger: The logger instance to use for logging
+//
+// Returns:
+//   - *AccountDefinition: A pointer to the account definition or nil if not found
+func (c *ConfigT) getAccountDefinition(accountDefinitionName string, accountAssociationLogger logger.Logger) *AccountDefinition {
+	if accountDefinitionName == "" {
+		return nil
+	}
+	if accountDefinition, exists := c.AccountDefinitions[accountDefinitionName]; exists {
+		return &accountDefinition
+	}
+	// Log error if account definition not found
+	accountAssociationLogger.Errorn("Account definition not found in configured accountDefinitions")
+	return nil
+}
+
+// populateAccountToDestination populates an account to a destination field based on the account ID.
+// It creates an Account object by combining account details with its definition.
+//
+// Parameters:
+//   - accountID: The ID of the account to populate
+//   - accountAssociationLogger: The logger instance to use for logging
+//
+// Returns:
+//   - *Account: The populated account object or nil if the account doesn't exist
+func (c *ConfigT) populateAccountToDestination(accountID string, accountAssociationLogger logger.Logger) *Account {
+	if account, exists := c.Accounts[accountID]; exists {
+		accountDefinitionPtr := c.getAccountDefinition(
+			account.AccountDefinitionName,
+			accountAssociationLogger,
+		)
+		account.AccountDefinition = accountDefinitionPtr
+		account.ID = accountID
+		return &account
+	}
+	accountAssociationLogger.Errorn("Account not found in configured accounts")
```

Comment:
Lets also print `accountId` for the same reason as previous

Reply:
It is already added to the logger itself. The log will show as below:
`2025-05-21T09:12:25.400+0530	ERROR	backend-config	backend-config/account_association.go:67	Account not found in configured accounts	{"destinationId": "2xO9bhFMegViLzux9VDWLCmsDAJ", "destinationType": "GOOGLE_ADWORDS_REMARKETING_LISTS", "rudderAccountId": "2xO9Z1yXE0Chr51OBFj0WVZbIm1", "flowType": "delivery"}`
Any other information do you think we need to show?

Reply:
I see this is fine